### PR TITLE
Fix: base repository with strict null checks

### LINKED ISSUE #443: 🤔 [ISSUE] - enabled strictNullChecks
### Description

We can notice that actually there no typing distinctions between undefined and null.

### Reproduction steps

```bash
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error
```

### Logs

_No response_

### Browsers

_No response_

### OS

_No response_

### PR DESCRIPTION
**Pull request by abdou6666**
# Motivation

This PR updates base-repository typing to work correctly in strictNullChecks mode.
 
Fixes #443 

# Type of change:

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: api/src/utils/generics/base-repository.ts [hunk 1]
```diff
@@ -166,33 +173,35 @@ export abstract class BaseRepository<
...
       const criteria = query.getFilter();
       const updates = query.getUpdate();
-
-      await repository.preUpdate(query, criteria, updates);
-      repository.emitter.emit(
-        repository.getEventName(EHook.preUpdate),
-        criteria,
-        updates?.['$set'],
-      );
+      if (updates) {
```
Comment (marrouchi) [inline api/src/utils/generics/base-repository.ts:180]: If the updates is undefined, I think we should throw an error.

File: api/src/utils/generics/base-repository.ts [hunk 2]
```diff
@@ -166,33 +173,35 @@ export abstract class BaseRepository<
...
       const criteria = query.getFilter();
       const updates = query.getUpdate();
-
-      await repository.preUpdateMany(query, criteria, updates);
-      repository.emitter.emit(
-        repository.getEventName(EHook.preUpdateMany),
-        criteria,
-        updates?.['$set'],
-      );
+      if (updates) {
```
Comment (marrouchi) [inline api/src/utils/generics/base-repository.ts:194]: Same here

File: api/src/utils/generics/base-repository.ts [hunk 3]
```diff
@@ -477,20 +486,21 @@ export abstract class BaseRepository<
     );
     const filterCriteria = query.getFilter();
     const queryUpdates = query.getUpdate();
+    if (queryUpdates) {
```
Comment (marrouchi) [inline api/src/utils/generics/base-repository.ts:489]: Same here

File: api/src/utils/pagination/pagination-query.dto.ts [hunk 4]
```diff
@@ -16,7 +16,7 @@ export type QuerySortDto<T> = [
 ];
 
 export type PageQueryDto<T> = {
-  skip: number | undefined;
-  limit: number | undefined;
-  sort: QuerySortDto<T>;
+  skip: number;
+  limit: number;
+  sort?: QuerySortDto<T>;
```
Comment (marrouchi) [inline api/src/utils/pagination/pagination-query.dto.ts:21]: Curious to know why we need to update this type.

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: api/src/utils/pagination/pagination-query.dto.ts (hunk 4 above)
Comment (abdou6666) [reply api/src/utils/pagination/pagination-query.dto.ts:21]: findAllQuery and findAll methods both have sort argument optional that's why

File: api/src/utils/generics/base-repository.ts [hunk 5]
```diff
@@ -179,11 +188,13 @@ export abstract class BaseRepository<
...
     });
 
-    hooks?.updateMany.pre.execute(async function () {
+    hooks.updateMany.pre.execute(async function () {
       const query = this as Query<D, D, unknown, T, 'updateMany'>;
       const criteria = query.getFilter();
       const updates = query.getUpdate();
-
+      if (!updates) {
+        throw new Error('Unable to run run updateMany pre hook');
```
Comment (marrouchi) [inline api/src/utils/generics/base-repository.ts:196]:
```suggestion
        throw new Error('Unable to execute updateMany() pre-hook');
```

File: api/src/utils/generics/base-repository.ts [hunk 6]
```diff
@@ -478,6 +489,10 @@ export abstract class BaseRepository<
     const filterCriteria = query.getFilter();
     const queryUpdates = query.getUpdate();
 
+    if (!queryUpdates) {
+      throw new Error('updateOne() query updates are not undefined');
```
Comment (marrouchi) [inline api/src/utils/generics/base-repository.ts:493]:
```suggestion
      throw new Error(' Unable to execute updateOne() - No updates');
```

File: api/src/utils/pagination/pagination-query.dto.ts (hunk 4 above)
Comment (marrouchi) [inline api/src/utils/pagination/pagination-query.dto.ts:21]:
```suggestion
  skip: number | undefined;
  limit: number | undefined;
  sort: QuerySortDto<T>;
```

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: api/src/utils/generics/base-repository.ts [hunk 7]
```diff
@@ -478,6 +489,10 @@ export abstract class BaseRepository<
     const filterCriteria = query.getFilter();
     const queryUpdates = query.getUpdate();
 
+    if (!queryUpdates) {
+      throw new Error(' Unable to execute updateOne() - No updates');
```
Comment (marrouchi) [inline api/src/utils/generics/base-repository.ts:493]:
```suggestion
      throw new Error('Unable to execute updateOne() - No updates');
```

**Review by marrouchi — CHANGES_REQUESTED** [review]

**Review by marrouchi — APPROVED** [review]

**Pull request merged**